Missing SCNReachability in Test Bundle

<!-- DO NOT DELETE
validate_template=true
template_path=.github/ISSUE_TEMPLATE/bug_report.md
-->
### Step 0: Are you in the right place?

  * For issues or feature requests related to __the code in this repository__
    file a Github issue.
    * If this is a __feature request__ please use the Feature Request template.
  * For general technical questions, post a question on [StackOverflow](http://stackoverflow.com/)
    with the `firebase` tag.
  * For general (non-iOS) Firebase discussion, use the [firebase-talk](https://groups.google.com/forum/#!forum/firebase-talk)
    google group.
  * For backend issues, console issues, and other non-SDK help that does not fall under one
    of the above categories, reach out to
    [Firebase Support](https://firebase.google.com/support/).
  * Once you've read this section and determined that your issue is appropriate for
    this repository, please delete this section.

### [REQUIRED] Step 1: Describe your environment

  * Xcode version: Version 12.0 beta 5 (12A8189h)
  * Firebase SDK version: 6.32-SPM-beta
  * Firebase Component: FireStore (Auth, Core, Database, Firestore, Messaging, Storage, etc)
  * Component version: 6.32
  * Installation method: SPM

### [REQUIRED] Step 2: Describe the problem
When running the test bundle I get:
```
Undefined symbols for architecture x86_64:
  "_SCNetworkReachabilityCreateWithAddress", referenced from:
      firebase::firestore::remote::(anonymous namespace)::CreateReachability() in SubFramework(FirebaseFirestore.o)
  "_SCNetworkReachabilitySetCallback", referenced from:
      firebase::firestore::remote::ConnectivityMonitorApple::ConnectivityMonitorApple(std::__1::shared_ptr<firebase::firestore::util::AsyncQueue> const&) in SubFramework(FirebaseFirestore.o)
  "_SCNetworkReachabilityGetFlags", referenced from:
      firebase::firestore::remote::ConnectivityMonitorApple::ConnectivityMonitorApple(std::__1::shared_ptr<firebase::firestore::util::AsyncQueue> const&) in SubFramework(FirebaseFirestore.o)
      firebase::firestore::remote::ConnectivityMonitorApple::OnEnteredForeground() in SubFramework(FirebaseFirestore.o)
  "_SCNetworkReachabilitySetDispatchQueue", referenced from:
      firebase::firestore::remote::ConnectivityMonitorApple::ConnectivityMonitorApple(std::__1::shared_ptr<firebase::firestore::util::AsyncQueue> const&) in SubFramework(FirebaseFirestore.o)
      firebase::firestore::remote::ConnectivityMonitorApple::~ConnectivityMonitorApple() in SubFramework(FirebaseFirestore.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
The project runs fine, but the test bundle will fail if a specific combination of files is used.

#### Steps to reproduce:
I'm not sure of the steps, but I have an example project. Where it occurs.
https://github.com/ryantstone/Firebase-Bug-Example

If a single test is commented out (`ConvertTests.test_timeToPickerValue_convertsHoursAndMinutes_toNativePickerData`) the test suite will build and run, but if that test is in it will fail.

Steps to reproduce are in the Readme in the repo.

--- a/FirebaseCore/CHANGELOG.md
+++ b/FirebaseCore/CHANGELOG.md
@@ -1,3 +1,7 @@
+# Firebase 6.33.0
+- [fixed] Swift Package Manager - Define system framework and system library dependencies. This
+  resolves undefined symbol issues for system dependencies. (#6408, #6413)
+
 # Firebase 6.32.0
 - [changed] Swift Package Manager - It's no longer necessary to select the Firebase or
   FirebaseCore products. Their build targets are implicitly selected when choosing any other

--- a/Package.swift
+++ b/Package.swift
@@ -148,6 +148,10 @@ let package = Package(
         .define("FIRCore_VERSION", to: firebaseVersion),
         .define("Firebase_VERSION", to: firebaseVersion),
         // TODO: - Add support for cflags cSetting so that we can set the -fno-autolink option
+      ],
+      linkerSettings: [
+        .linkedFramework("UIKit", .when(platforms: .some([.iOS, .tvOS]))),
+        .linkedFramework("AppKit", .when(platforms: .some([.macOS]))),
       ]
     ),
     .testTarget(
@@ -245,6 +249,10 @@ let package = Package(
         .headerSearchPath("../../"),
         .define("FIRAuth_VERSION", to: firebaseVersion),
         .define("FIRAuth_MINOR_VERSION", to: "1.1"), // TODO: Fix version
+      ],
+      linkerSettings: [
+        .linkedFramework("Security"),
+        .linkedFramework("SafariServices", .when(platforms: .some([.iOS]))),
       ]
     ),
     .testTarget(
@@ -295,6 +303,10 @@ let package = Package(
         .define("PB_FIELD_32BIT", to: "1"),
         .define("PB_NO_PACKED_STRUCTS", to: "1"),
         .define("PB_ENABLE_MALLOC", to: "1"),
+      ],
+      linkerSettings: [
+        .linkedFramework("Security"),
+        .linkedFramework("SystemConfiguration"),
       ]
     ),
 
@@ -315,6 +327,11 @@ let package = Package(
       cSettings: [
         .headerSearchPath("../../"),
         .define("FIRDatabase_VERSION", to: firebaseVersion),
+      ],
+      linkerSettings: [
+        .linkedFramework("CFNetwork"),
+        .linkedFramework("Security"),
+        .linkedFramework("SystemConfiguration"),
       ]
     ),
     .testTarget(
@@ -340,6 +357,9 @@ let package = Package(
         .define("FIRDynamicLinks3P", to: "1"),
         .define("GIN_SCION_LOGGING", to: "1"),
         .define("FIRDynamicLinks_VERSION", to: firebaseVersion),
+      ],
+      linkerSettings: [
+        .linkedFramework("QuartzCore"),
       ]
     ),
 
@@ -491,6 +511,9 @@ let package = Package(
       publicHeadersPath: "Public",
       cSettings: [
         .headerSearchPath("../../../"),
+      ],
+      linkerSettings: [
+        .linkedFramework("Security"),
       ]
     ),
     .target(
@@ -549,6 +572,10 @@ let package = Package(
       cSettings: [
         .headerSearchPath("../../"),
         .define("FIRStorage_VERSION", to: firebaseVersion),
+      ],
+      linkerSettings: [
+        .linkedFramework("MobileCoreServices", .when(platforms: .some([.iOS]))),
+        .linkedFramework("CoreServices", .when(platforms: .some([.macOS]))),
       ]
     ),
     .testTarget(

--- a/Package.swift
+++ b/Package.swift
@@ -586,6 +586,10 @@ let package = Package(
         .define("PB_FIELD_32BIT", to: "1"),
         .define("PB_NO_PACKED_STRUCTS", to: "1"),
         .define("PB_ENABLE_MALLOC", to: "1"),
+      ],
+      linkerSettings: [
+        .linkedFramework("SystemConfiguration"),
+        .linkedFramework("CoreTelephony", .when(platforms: .some([.macOS, .iOS])))
       ]
     ),
     .testTarget(

--- a/Package.swift
+++ b/Package.swift
@@ -623,7 +623,7 @@ let package = Package(
       ],
       linkerSettings: [
         .linkedFramework("SystemConfiguration"),
-        .linkedFramework("CoreTelephony", .when(platforms: .some([.macOS, .iOS])))
+        .linkedFramework("CoreTelephony", .when(platforms: .some([.macOS, .iOS]))),
       ]
     ),
     .testTarget(

--- a/Package.swift
+++ b/Package.swift
@@ -297,6 +297,7 @@ let package = Package(
         .define("PB_ENABLE_MALLOC", to: "1"),
       ]
     ),
+
     .target(
       name: "FirebaseDatabase",
       dependencies: [
@@ -398,6 +399,12 @@ let package = Package(
         .define("PB_NO_PACKED_STRUCTS", to: "1"),
         .define("PB_ENABLE_MALLOC", to: "1"),
         .define("FIRFirestore_VERSION", to: firebaseVersion),
+      ],
+      linkerSettings: [
+        .linkedFramework("SystemConfiguration"),
+        .linkedFramework("MobileCoreServices", .when(platforms: .some([.iOS]))),
+        .linkedFramework("UIKit", .when(platforms: .some([.iOS, .tvOS]))),
+        .linkedLibrary("c++"),
       ]
     ),
     .target(

